lib/zclient.c

```diff
@@ -2434,36 +2434,44 @@ size_t zebra_interface_link_params_write(struct stream *s,
 	size_t w;
 	struct if_link_params *iflp;
 	int i;
+	struct in_addr inaddr_none_ip;
+
+	inaddr_none_ip.s_addr = INADDR_NONE;
 
-	if (s == NULL || ifp == NULL || ifp->link_params == NULL)
+	if (s == NULL || ifp == NULL)
 		return 0;
 
 	iflp = ifp->link_params;
+
 	w = 0;
 
-	w += stream_putl(s, iflp->lp_status);
+	w += stream_putc(s, iflp ? iflp->enabled : 0);
+	w += stream_putl(s, iflp ? iflp->lp_status : 0);
 
-	w += stream_putl(s, iflp->te_metric);
-	w += stream_putf(s, iflp->max_bw);
-	w += stream_putf(s, iflp->max_rsv_bw);
+	w += stream_putl(s, iflp ? iflp->te_metric : 0);
+	w += stream_putf(s, iflp ? iflp->max_bw : 0);
+	w += stream_putf(s, iflp ? iflp->max_rsv_bw : 0);
 
-	w += stream_putl(s, MAX_CLASS_TYPE);
-	for (i = 0; i < MAX_CLASS_TYPE; i++)
-		w += stream_putf(s, iflp->unrsv_bw[i]);
+	if (iflp) {
+		w += stream_putl(s, iflp ? MAX_CLASS_TYPE : 0);
+		for (i = 0; i < MAX_CLASS_TYPE; i++)
+			w += stream_putf(s, iflp->unrsv_bw[i]);
+	} else
+		w += stream_putl(s, 0);
 
-	w += stream_putl(s, iflp->admin_grp);
-	w += stream_putl(s, iflp->rmt_as);
-	w += stream_put_in_addr(s, &iflp->rmt_ip);
+	w += stream_putl(s, iflp ? iflp->admin_grp : 0);
+	w += stream_putl(s, iflp ? iflp->rmt_as : 0);
+	w += stream_put_in_addr(s, iflp ? &iflp->rmt_ip : &inaddr_none_ip);
 
```

Comment:
inaddr_any is a global variable, I guess

Reply:
code simplifié